core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -51,6 +51,7 @@ import org.apache.spark.util.Utils
 private[spark] class NettyBlockTransferService(
     conf: SparkConf,
     securityManager: SecurityManager,
+    serializerManager: SerializerManager,
```

Comment:
The reason for using the `serializerManager` is that we get a configured serializer that are aware of custom jars. It being configured here https://github.com/apache/spark/blob/3cdbc176337bdb0dbe11a120b0416b4f9014d8d9/core/src/main/scala/org/apache/spark/executor/Executor.scala#L168-L172 so we are just constructing a `JavaSerializer` internally we need to expose some method so we can configure the classloader that get used by it.

Reply:
That config comes through conf right? I am just wondering what the difference is if you still make the JavaSerializer and serializer manager inside the class rather than make the caller do it now?

Reply:
I think I'm asking something narrower, and maybe still misinformed.

Right now you've basically lifted "new JavaSerializer(conf)" into the caller, where it's wrapped in a SerializerManager, then passed back in. NettyBlockTransferService, for instance, already takes 'conf' as an argument. What's different if you make the JavaSerializer and SeralizerManager inside the class, from 'conf'? I'm just missing how it's different.

I see that this isn't quite the case in a few tests though, OK. Maybe that's reason enough for this setup, OK.